### PR TITLE
Option to toggle display of column hide ::[ Table widget ] 

### LINKED ISSUE #4481: Option to toggle display  of column hide ::[ Table widget ]
### Is your proposal related to a problem?

Currently no option to hide disable column icon

### Describe the solution you'd like

Add property to toggle show/hide of column hide option.

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by stepinfwd**
closes #4481

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2x6h5fg Option to switch on/off show hide button : Table widget](https://app.clickup.com/t/2x6h5fg)

File: frontend/src/Editor/WidgetManager/widgetConfig.js [hunk 1]
```diff
@@ -162,6 +162,13 @@ export const widgets = [
           schema: { type: 'boolean' },
         },
       },
+      toggleShowColumns: {
+        type: 'toggle',
+        displayName: 'Toggle hide columns',
```
Comment (sherfin94) [inline frontend/src/Editor/WidgetManager/widgetConfig.js:167]:
```suggestion
        displayName: 'Hide column selector button',
```

File: frontend/src/Editor/Components/Table/load-properties-and-styles.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ export default function loadPropertiesAndStyles(properties, styles, darkMode, co
 
   const totalRecords = properties.totalRecords ?? '';
   const enabledSort = properties?.enabledSort ?? true;
+  const toggleShowColumns = properties?.toggleShowColumns ?? false;
```
Comment (sherfin94) [inline frontend/src/Editor/Components/Table/load-properties-and-styles.js:13]:
```suggestion
  const hideColumnSelectorButton = properties?.hideColumnSelectorButton ?? false;
```
Let's rename likewise everywhere else also

File: frontend/src/Editor/Components/Table/Table.jsx [hunk 3]
```diff
@@ -600,51 +601,57 @@ export function Table({
...
-                              />
-                              <span
-                                className="hide-column-name"
-                                data-cy={`options-coloumn-${String(column.Header).toLowerCase().replace(/\s+/g, '-')}`}
-                              >
-                                {` ${column.Header}`}
-                              </span>
-                            </label>
-                          </div>
+              {toggleShowColumns && (
```
Comment (sherfin94) [inline frontend/src/Editor/Components/Table/Table.jsx:604]:
```suggestion
              {!hideColumnSelectorButton && (
```
This negation is required given the new property naming.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Please change the property name and corresponding variable names as suggested. Let's not keep misleading names.

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**